gears/src/commands/client/tx.rs

```diff
@@ -39,31 +85,78 @@ pub struct LocalInfo {
 }
 
 pub fn run_tx<C, H: TxHandler<TxCommands = C>>(
-    TxCommand {
-        node,
-        chain_id,
-        fees,
-        inner,
-        keyring,
-    }: TxCommand<C>,
+    command: TxCommand<C>,
     handler: &H,
-) -> anyhow::Result<Response> {
-    match keyring {
+) -> anyhow::Result<Vec<Response>> {
+    match command.keyring {
         Keyring::Ledger => {
             let key = LedgerProxyKey::new()?;
 
-            let message = handler.prepare_tx(inner, key.get_address())?;
-            handler.handle_tx(message, key, node, chain_id, fees, SignMode::Textual)
+            let ctx = &(&command).into();
+            let messages = handler.prepare_tx(ctx, command.inner, key.get_address())?;
+            // TODO: maybe remove error and replace with warning message
+            if messages.chunk_size() != 0 {
+                return Err(anyhow!(
+                    "ledger device cannot create more that one transaction per command run"
+                ));
+            }
+            handler
+                .handle_tx(
+                    messages,
+                    key,
+                    command.node,
+                    command.chain_id,
+                    command.fees,
+                    SignMode::Textual,
+                )
+                .map(|res| vec![res])
         }
-        Keyring::Local(info) => {
+        Keyring::Local(ref info) => {
             let keyring_home = info.home.join(info.keyring_backend.get_sub_dir());
             let key = keyring::key_by_name(
                 &info.from_key,
                 info.keyring_backend.to_keyring_backend(&keyring_home),
             )?;
 
-            let message = handler.prepare_tx(inner, key.get_address())?;
-            handler.handle_tx(message, key, node, chain_id, fees, SignMode::Direct)
+            let ctx = &(&command).into();
+            let messages = handler.prepare_tx(ctx, command.inner, key.get_address())?;
+
+            if messages.chunk_size() > 0
+            /* && command.broadcast_mode == BroadcastMode::Block */
+            {
+                let step = messages.chunk_size() as usize;
+                let msgs = messages.into_msgs();
+
+                let mut res = vec![];
+                for i in (0..msgs.len()).step_by(step) {
+                    res.push(
+                        handler.handle_tx(
+                            msgs[i..(i + step).min(msgs.len())]
```

Comment:
Would [this](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.chunks) be useful here?

Reply:
It's exactly what I've searched in editor popup tips and didn't find.